editor/group_settings_editor.cpp

```diff
@@ -0,0 +1,496 @@
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "group_settings_editor.h"
+
+#include "core/config/project_settings.h"
+#include "editor/editor_scale.h"
+#include "editor/editor_undo_redo_manager.h"
+#include "editor/scene_tree_dock.h"
+#include "editor_file_system.h"
+#include "editor_node.h"
+#include "scene/resources/packed_scene.h"
+
+void GroupSettingsEditor::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE: {
+			update_tree();
+		} break;
+	}
+}
+
+HashMap<StringName, String> GroupSettingsEditor::get_groups() const {
+	return _groups_cache;
+}
+
+void GroupSettingsEditor::_item_edited() {
+	if (updating_group) {
+		return;
+	}
+
+	TreeItem *ti = tree->get_edited();
+
+	if (!ti) {
+		return;
+	}
+
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	if (tree->get_edited_column() == 0) {
+		// Name Edited
+		String new_name = ti->get_text(0);
+		String old_name = ti->get_meta("__name");
+
+		if (new_name == old_name) {
+			return;
+		}
+
+		String error = _check_new_group_name(new_name);
+		if (!error.is_empty()) {
+			ti->set_text(0, old_name);
+			show_message(error);
+			return;
+		}
+
+		undo_redo->create_action(TTR("Rename Group"));
+
+		undo_redo->add_do_method(this, "_rename_group", old_name, new_name);
+		undo_redo->add_undo_method(this, "_rename_group", new_name, old_name);
+
+		undo_redo->add_do_method(this, "_rename_references", old_name, new_name);
+		undo_redo->add_undo_method(this, "_rename_references", new_name, old_name);
+
+		undo_redo->add_do_method(this, "call_deferred", "update_tree");
+		undo_redo->add_undo_method(this, "call_deferred", "update_tree");
+
+		undo_redo->add_do_method(this, "emit_signal", group_changed);
+		undo_redo->add_undo_method(this, "emit_signal", group_changed);
+
+		undo_redo->add_do_method(this, "_save_groups");
+		undo_redo->add_undo_method(this, "_save_groups");
+
+		undo_redo->commit_action();
+	} else if (tree->get_edited_column() == 1) {
+		// Description Edited
```

Comment:
```suggestion
		// Description Edited.
```
The code style requires comments to start with a capital and end with a period, same everywhere